dandiapi/api/views/dandiset.py

```diff
@@ -163,47 +164,59 @@ def get_object(self):
     def list(self, request, *args, **kwargs):
         qs = self.get_queryset()
         dandisets = self.paginate_queryset(self.filter_queryset(qs))
-
-        # Query versions, to supply size/count of assets.
-        base_query = (
+        relevant_versions = (
             Version.objects.select_related('dandiset')
-            .annotate(
-                num_assets=Count('assets'),
-                total_size=(
-                    Coalesce(Sum('assets__blob__size'), 0)
-                    + Coalesce(Sum('assets__embargoed_blob__size'), 0)
-                    + Coalesce(Sum('assets__zarr__size'), 0)
-                ),
-            )
             .filter(dandiset__in=dandisets)
             .order_by('-version', '-modified')
         )
 
+        # Get all published versions
         latest_dandiset_version = (
             Version.objects.exclude(version='draft')
             .order_by('-version')
             .filter(dandiset_id=OuterRef('dandiset_id'))
             .values('version')[:1]
         )
         published = (
-            base_query.exclude(version='draft')
+            relevant_versions.exclude(version='draft')
             .alias(latest=Subquery(latest_dandiset_version))
             .filter(version=F('latest'))
         )
-        drafts = base_query.filter(version='draft')
 
-        # Union with drafts
+        # Get all draft versions
+        drafts = relevant_versions.filter(version='draft')
+
+        # Union published with drafts
         versions = published.union(drafts).order_by('dandiset_id', '-version')
 
-        # Organize draft and most recently published for each dandiset.
+        # Map version IDs to their stats
+        version_stats = {}
+        root_paths = get_root_paths_many(versions=relevant_versions)
+        for path in root_paths:
```

Comment:
Won't introducing this loop over this queryset makes this endpoint's run time grow linearly with the number of root-level assets in each dandiset? i.e. if someone uploads a dandiset with 10,000 assets at the root level, will that add 10,000 iterations to this loop?

Reply:
I believe that's correct, but it's not a problem due to the following:

1. While it may be linear, the "coefficient" of that is _extremely low_, since we're dealing with simple postgres queries
2. Based on the data we've seen so far, it's unlikely for a dandiset to contain 10,000 assets at the root level (most I've seen is ~1100). In fact, at the time of writing this, the entire production archive has barely more than 10,000 root paths across all versions
3. Even if we run this loop over ~10k root paths, it takes ~200ms to run, which I think is acceptable, and certainly better than what we had

So for those reasons, I don't see a reason to be concerned about this yet.

Reply:
That makes sense to me 👍